src/DIRAC/WorkloadManagementSystem/Agent/JobAgent.py

```diff
@@ -729,6 +729,12 @@ def finalize(self):
     """ Job Agent finalization method
     """
 
+    res = self.computingElement.shutdown()
```

Comment:
And maybe another comment here to say that `shutdown()` will force the `JobAgent` to wait until all the tasks are in a final state?